### PR TITLE
Timezones list renewal in Scheduler

### PR DESCRIPTION
**Pull request by Yarovaya**
- renovation of the timezones list according to 2020 data
- render only one timeZone editor for appointment timeZone fields

### CODE REVIEW

File: js/ui/scheduler/timezones/ui.scheduler.timezones.js [hunk 1]
```diff
@@ -1,20 +1,45 @@
...
 
     getTimezones: function() {
-        return this._list;
+        return this._timeZones;
     },
-    getDisplayNames: function() {
-        return this._displayNames;
+
+    getDisplayedTimeZones: function(date) {
+        date = date || new Date();
```
Comment (AryamnovEugeniy) [inline js/ui/scheduler/timezones/ui.scheduler.timezones.js]: I think it is better to create a const here

File: testing/tests/DevExpress.ui.widgets.scheduler/appointmentPopup.tests.js [hunk 2]
```diff
@@ -1716,3 +1629,166 @@ QUnit.test('Popup should not be closed until the valid value is typed', function
...
+        assert.notOk(startDateTimezoneEditor, 'StartDateTZ editor isn\'t visible by default');
+    });
+
+    test('Popup should not contain endDateTimeZone editor by default', function(assert) {
+        this.instance.showAppointmentPopup({ startDate: new Date(2015, 1, 1, 1), endDate: new Date(2015, 1, 1, 2), text: 'caption', description: 'First task of this day', allDay: true });
+
+        const form = this.instance.getAppointmentDetailsForm();
+        const endDateTimeZoneEditor = form.getEditor('endDateTimeZone');
+
+        assert.notOk(endDateTimeZoneEditor, 'StartDateTZ editor isn\'t visible by default');
```
Comment (AryamnovEugeniy) [inline testing/tests/DevExpress.ui.widgets.scheduler/appointmentPopup.tests.js]: EndDateTZ

File: js/ui/scheduler/appointmentPopup.js [hunk 3]
```diff
@@ -211,28 +211,14 @@ export default class AppointmentPopup {
...
-        const { startDateExpr, endDateExpr, recurrenceRuleExpr } = this.scheduler._dataAccessors.expr;
-
-        const recurrenceEditorOptions = this._getEditorOptions(recurrenceRuleExpr, APPOINTMENT_FORM_GROUP_NAMES.Recurrence);
-        this._setEditorOptions(recurrenceRuleExpr, APPOINTMENT_FORM_GROUP_NAMES.Recurrence, extend({}, recurrenceEditorOptions, { startDate: startDate }));
+        const { startDateExpr, endDateExpr } = this.scheduler._dataAccessors.expr;
 
         this._appointmentForm.option('readOnly', this._isReadOnly(data));
 
-        AppointmentForm.updateFormData(this._appointmentForm, formData);
+        AppointmentForm.updateFormData(this._appointmentForm, formData, this.scheduler._dataAccessors.expr, startDate, endDate);
```
Comment (AntonSermyazhko) [inline js/ui/scheduler/appointmentPopup.js]: A lot of arguments.
Is it possible to use `this.scheduler._dataAccessors.expr` inside `updateFormData` method?
Does `formData` not contains `startDate`, `endDate`?

File: js/ui/scheduler/timezones/ui.scheduler.timezones.js (hunk 1 above)
Comment (Yarovaya) [reply js/ui/scheduler/timezones/ui.scheduler.timezones.js]: @AryamnovEugeniy  this code has no effect in the latest version of code in PR,  so I'll just give up this line

File: js/ui/scheduler/appointmentPopup.js (hunk 3 above)
Comment (Yarovaya) [reply js/ui/scheduler/appointmentPopup.js]: @AntonSermyazhko There are some difficulties with refactoring there. ui.scheduler.appointmentForm is a kind of facade above Form.  This entity does not have access to compiled getters to access fields. But I will try to refactor

**Review by AntonSermyazhko — APPROVED** [review]

**Review by andrewmakarov — APPROVED** [review]

**Review by AryamnovEugeniy — APPROVED** [review]

**Pull request merged**